### PR TITLE
Sockets: fix `getipaddr()`

### LINKED ISSUE #56520: Julia on servers without IPv4
I am trying to build julia 1.11.1 on a server that only has IPv6 interfaces. I was able to build the julia binary but the networking related tests were skipped. And, they were skipped because `getipaddr()` throws an exception:

```julia
julia> getipaddr()
ERROR: No networking interface available
```

This is where it's invoked from the test:
https://github.com/JuliaLang/julia/blob/8f5b7ca12ad48c6d740e058312fc8cf2bbe67848/test/choosetests.jl#L188-L202

I do have IPv6 addresses:
```
julia> getipaddrs()
3-element Vector{IPAddr}:
...
```

The documentation suggest that it should return an IP address, just having a preference of IPv4 over IPv6. 
```julia
help?> getipaddr
search: getipaddr getipaddrs getpid getindex getalladdrinfo

  getipaddr() -> IPAddr

  Get an IP address of the local machine, preferring IPv4 over IPv6. Throws if no addresses are available.

  getipaddr(addr_type::Type{T}) where T<:IPAddr -> T

  Get an IP address of the local machine of the specified type. Throws if no addresses of the specified type are available.

  This function is a backwards-compatibility wrapper around getipaddrs. New applications should use getipaddrs instead.

  Examples
  ≡≡≡≡≡≡≡≡

  julia> getipaddr()
  ip"192.168.1.28"
  
  julia> getipaddr(IPv6)
  ip"fe80::9731:35af:e1c5:6e49"

  See also getipaddrs.
```

So, why doesn't getipaddr() return IPv6 address instead? Do you consider this a bug or an error in documentation? I feel that it's the former.

### PR DESCRIPTION
**Pull request by barucden**
This PR updates the definition of `getipaddr()` so that the documentation is respected.

Before this change, `getipaddr()` would throw an error unless there is an `IPv4` address assigned to a network interface. After this change, `getipaddr()` preferably returns an `IPv4` address, but if none exists, it returns an `IPv6` address.

Fixes #56520

### CODE REVIEW

**Comment by martinholters** [comment]
IIUC, the current behavior was deliberately chosen in #30609 to avoid breakage in code that has problems with IPv6. Not sure that's a valid concern still.

That said, isn't
```diff
-getipaddr() = getipaddr(IPv4)
+getipaddr() = getipaddr(IPAddr)
```
all that's needed as a change here?

**Comment by barucden** [comment]
Thank you for your input. I would have never guessed that `IPAddr` is a valid input for that function! But you are right, your proposal would already fix the linked issue.

So is it correct that the whole `findfirst` logic is there so that `getaddrs(IPAddr)` returns an IPv4?

**Comment by martinholters** [comment]
> So is it correct that the whole `findfirst` logic is there so that `getaddrs(IPAddr)` returns an IPv4?

I think so, yes.

File: stdlib/Sockets/src/addrinfo.jl [hunk 1]
```diff
@@ -282,16 +282,14 @@ See also [`getipaddrs`](@ref).
 """
 function getipaddr(addr_type::Type{T}) where T<:IPAddr
     addrs = getipaddrs(addr_type)
+    isempty(addrs) || error("No networking interface available")
```
Comment (vtjnash) [inline stdlib/Sockets/src/addrinfo.jl:285]:
```suggestion
    isempty(addrs) && error("No networking interface available")
```

**Comment by vtjnash** [comment]
Yes, looks like the findfirst logic was added with the intent of making this work this way, but forgot to make the change to getipaddr (https://github.com/JuliaLang/julia/issues/32179#issuecomment-497162388)

**Pull request merged**